spark_log_parser/parsing_models/exceptions.py

```diff
@@ -20,84 +21,106 @@ def __init__(
+        """
+        # Convert into a pydantic model in future iterations
+        # once flask is stripped away
+        return {
+            "error": {
+                "type": self.error_type,
+                "code": self.status_code,
+                "message": self.error_message,
+            }
+        }
 
 
+class ConfigurationException(SyncParserException):
+    def __init__(self, config_recs: str):
+
         error_message = ParserErrorMessages.SPARK_CONFIG_GENERIC_MESSAGE
 
         for idx, c in enumerate(config_recs):
-            count = idx+1
-            error_message += f'  ({count}) {c}'
+            count = idx + 1
+            error_message += f"  ({count}) {c}"
 
-        error_message += f'. {ParserErrorMessages.SUPPORT_MESSAGE}'
+        error_message += f". {ParserErrorMessages.SUPPORT_MESSAGE}"
 
         super().__init__(
             error_type=ParserErrorTypes.SPARK_CONFIG_ERROR,
             error_message=error_message,
             status_code=ParserErrorCodes.SPARK_CONFIG_ERROR,
-            exception=None)
+        )
+
 
 class LazyEventValidationException(SyncParserException):
     """
     This Exception is for missing event data that doesn't immediately kill the parser.
     All of the related missing events can be gathered and identified in the error message.
     """
+
     def __init__(self, error_message: str):
 
         error_message += (
-            f"{ParserErrorMessages.MISSING_EVENT_EXPLANATION} " +
-            f"{ParserErrorMessages.SUPPORT_MESSAGE}")
+            f"{ParserErrorMessages.MISSING_EVENT_EXPLANATION} "
+            + f"{ParserErrorMessages.SUPPORT_MESSAGE}"
+        )
 
         super().__init__(
             error_type=ParserErrorTypes.MISSING_EVENT_ERROR,
             error_message=error_message,
             status_code=ParserErrorCodes.SPARK_EVENT_ERROR,
-            exception=None)
+        )
 
 
 class UrgentEventValidationException(SyncParserException):
     """
     This Exception is for missing event data that stops the parser dead in its tracks.
     """
-    def __init__(self, missing_event: str = ''):
+
+    def __init__(self, missing_event: str = ""):
 
         error_message = (
-            f"{ParserErrorMessages.MISSING_EVENT_GENERIC_MESSAGE} '{missing_event}'. " +
-            f"{ParserErrorMessages.MISSING_EVENT_EXPLANATION} " +
-            f"{ParserErrorMessages.SUPPORT_MESSAGE}")
+            f"{ParserErrorMessages.MISSING_EVENT_GENERIC_MESSAGE} '{missing_event}'. "
+            + f"{ParserErrorMessages.MISSING_EVENT_EXPLANATION} "
+            + f"{ParserErrorMessages.SUPPORT_MESSAGE}"
+        )
 
         super().__init__(
             error_type=ParserErrorTypes.MISSING_EVENT_ERROR,
             error_message=error_message,
             status_code=ParserErrorCodes.SPARK_EVENT_ERROR,
-            exception=None)
+        )
+
+
+class LogSubmissionException(SyncParserException):
+    """
+    This Exception is for malformed log submission
+    """

```

Comment:
Since these are also ValueErrors it'd be good to inherit from that too (e.g. https://github.com/pydantic/pydantic/blob/c22fe9a8f379fc0342345109f7451d98dc433a2d/pydantic/error_wrappers.py#L50), tying this in to the hierarchy: https://docs.python.org/3/library/exceptions.html#exception-hierarchy